### PR TITLE
fix: make all bash script pass shellcheck

### PR DESCRIPTION
**Pull request by leseb**
All our bash scripts are now passing the shellcheck linter.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Review by hickeyma — COMMENTED** [review]
@leseb Is this PR related to #933?

**Comment by leseb** [comment]
> @leseb Is this PR related to #933?

Somehow but no direct connection. This PR cleans up all the warnings. The other one was partial and I think #933 could stick with the CI job only. Thanks!

**Review by hickeyma — COMMENTED** [review]
> Somehow but no direct connection. This PR cleans up all the warnings. The other one was partial and I think https://github.com/instructlab/instructlab/pull/933 could stick with the CI job only. Thanks!

My thinking is that on adding checker/linter is a follows:

1. Add checker to code base
2. Enable checker and fix all suggestions

When you merge 1 you then merge 2. This way you have a base where going forward the checker is in place, so any PR afterwards knows it is introducing the checker errors.

Will merging this PR just fix the shell checker errors for this moment in time? If so, what about the next person who makes changes/adds scripts and adds checker errors?

**Comment by leseb** [comment]
> > Somehow but no direct connection. This PR cleans up all the warnings. The other one was partial and I think #933 could stick with the CI job only. Thanks!
> 
> My thinking is that on adding checker/linter is a follows:
> 
> 1. Add checker to code base
> 2. Enable checker and fix all suggestions
> 
> When you merge 1 you then merge 2. This way you have a base where going forward the checker is in place, so any PR afterwards knows it is introducing the checker errors.
> 
> Will merging this PR just fix the shell checker errors for this moment in time? If so, what about the next person who makes changes/adds scripts?

Sorry for the confusion, my thinking is that this PR should be merged since it fixes all the shellcheck issues, then #933 should follow so that any changes to bash scripts will be lint.
Does that make sense? Thanks!

I commented on https://github.com/instructlab/instructlab/pull/933#issuecomment-2134972953 let's wait a few, if no response I'll just cherry-pick https://github.com/instructlab/instructlab/pull/933/commits/f1148a0edf169f9d12637869392fb1f89fadc618 in this PR.

**Comment by hickeyma** [comment]
> Sorry for the confusion, my thinking is that this PR should be merged since it fixes all the shellcheck issues, then https://github.com/instructlab/instructlab/pull/933 should follow so that any changes to bash scripts will be lint.

If you do it this way, you are waiting on the tool to be added and enabled. My suggestions is to follow the approach in https://github.com/instructlab/instructlab/issues/77 where tools are added first and then PR follows with tool enabled in CI/CD containing all changes in relevant files as suggested by the tool. This way you are enabled and ready to go.

**Comment by leseb** [comment]
> @leseb I see that #933 is closed and not being continued. Does this mean we run shellcheck on the fly and update changes like this PR? Or are we going to add it to CI/CD?

I'm writing the CI job at the moment and will push shortly an update to this PR.

**Comment by leseb** [comment]
Simulated failure can be seen [here](https://github.com/instructlab/instructlab/actions/runs/9285708013/job/25550833963?pr=1211)

Screenshot 2024-05-29 at 14 04 58

File: .github/workflows/shellcheck.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - "main"
+      - "release-**"
+    paths:
+      - '**.sh'
+  pull_request:
+    branches:
+      - "main"
+      - "release-**"
+    paths:
+      - '**.sh'
```
Comment (russellb) [inline .github/workflows/shellcheck.yaml:17]: How about when this file is changed as well?

**Review by russellb — APPROVED** [review]
one minor suggestion, but lgtm!

File: .github/workflows/shellcheck.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+name: Shellcheck
+
+on:
+  push:
+    branches:
+      - "main"
+      - "release-**"
+    paths:
+      - '**.sh'
```
Comment (bjhargrave) [inline .github/workflows/shellcheck.yaml:11]:
```suggestion
      - '**.sh'
      - '.github/workflows/shellcheck.yaml' # This workflow
```

File: .github/workflows/shellcheck.yaml (hunk 1 above)
Comment (bjhargrave) [inline .github/workflows/shellcheck.yaml:17]:
```suggestion
      - '**.sh'
      - '.github/workflows/shellcheck.yaml' # This workflow
```

File: .github/workflows/shellcheck.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  pull_request:
+    branches:
+      - "main"
+      - "release-**"
+    paths:
+      - '**.sh'
+
+permissions:
```
Comment (bjhargrave) [inline .github/workflows/shellcheck.yaml:19]:
```suggestion
env:
  LC_ALL: en_US.UTF-8

defaults:
  run:
    shell: bash


permissions:
```

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: .github/workflows/shellcheck.yaml (hunk 3 above)
Comment (leseb) [reply .github/workflows/shellcheck.yaml:19]: Why do we need?

```
defaults:
	  run:
	    shell: bash
```

Comment (leseb) [reply .github/workflows/shellcheck.yaml:21]: Actually, why do we need all the suggestions? Thanks!

**Comment by leseb** [comment]
Need https://github.com/instructlab/instructlab/pull/1246 for CI to be green

**Comment by russellb** [comment]
It looks like @hickeyma 's comments were addressed, so I'm going to drop that pending review and let this merge. Let me know if I missed anything, though!

**Pull request merged**